Problem compiling

Apologies as this is probably a basic Node.js/npm question or something but I'm totally new to these things and wanted to try out master, 

I followed the instructions and cloned the repo, then did `npm install` just fine, but `npm run build:all` gives,

```
~/src/jupyterlab_app$ npm run build:all

> jupyterlab_app@0.1.1 build:all /home/marius/src/jupyterlab_app
> npm run build && npm start


> jupyterlab_app@0.1.1 build /home/marius/src/jupyterlab_app
> tsc && node ./scripts/extract.js && node ./scripts/copyassets.js && npm run build:pack

node_modules/@types/react-dom/node_modules/@types/react/index.d.ts(3420,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'a' must be of type 'DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>', but here has type 'DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>'.

#
# about 100 other similar lines
#

npm ERR! Linux 4.4.0-96-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "build"
npm ERR! node v6.11.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! jupyterlab_app@0.1.1 build: `tsc && node ./scripts/extract.js && node ./scripts/copyassets.js && npm run build:pack`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jupyterlab_app@0.1.1 build script 'tsc && node ./scripts/extract.js && node ./scripts/copyassets.js && npm run build:pack'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the jupyterlab_app package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tsc && node ./scripts/extract.js && node ./scripts/copyassets.js && npm run build:pack
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs jupyterlab_app
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls jupyterlab_app
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/marius/src/jupyterlab_app/npm-debug.log

npm ERR! Linux 4.4.0-96-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "build:all"
npm ERR! node v6.11.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! jupyterlab_app@0.1.1 build:all: `npm run build && npm start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jupyterlab_app@0.1.1 build:all script 'npm run build && npm start'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the jupyterlab_app package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build && npm start
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs jupyterlab_app
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls jupyterlab_app
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/marius/src/jupyterlab_app/npm-debug.log
```

Any hints what I might be doing wrong?

Thanks. Upgraded to Node.js v8.6.0 and npm 5.3.0, as well as fixed a error about node-gyp / python3 that I had missed happened in the `npm install`. 

Now the bottom of the output after `node run build:all` looks like, 

```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jupyterlab_app@0.1.1 build: `tsc && node ./scripts/extract.js && node ./scripts/copyassets.js && npm run build:pack`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jupyterlab_app@0.1.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/marius/.npm/_logs/2017-10-05T22_41_30_384Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! jupyterlab_app@0.1.1 build:all: `npm run build && npm start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the jupyterlab_app@0.1.1 build:all script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/marius/.npm/_logs/2017-10-05T22_41_30_421Z-debug.log
```

and the contents of that log is,

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build:all' ]
2 info using npm@5.3.0
3 info using node@v8.6.0
4 verbose run-script [ 'prebuild:all', 'build:all', 'postbuild:all' ]
5 info lifecycle jupyterlab_app@0.1.1~prebuild:all: jupyterlab_app@0.1.1
6 info lifecycle jupyterlab_app@0.1.1~build:all: jupyterlab_app@0.1.1
7 verbose lifecycle jupyterlab_app@0.1.1~build:all: unsafe-perm in lifecycle true
8 verbose lifecycle jupyterlab_app@0.1.1~build:all: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/home/marius/src/jupyterlab_app/node_modules/.bin:/home/marius/workspace/clik-hg/bin:/home/marius/bin:/home/marius/.local/bin:/home/marius/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
9 verbose lifecycle jupyterlab_app@0.1.1~build:all: CWD: /home/marius/src/jupyterlab_app
10 silly lifecycle jupyterlab_app@0.1.1~build:all: Args: [ '-c', 'npm run build && npm start' ]
11 silly lifecycle jupyterlab_app@0.1.1~build:all: Returned: code: 1  signal: null
12 info lifecycle jupyterlab_app@0.1.1~build:all: Failed to exec build:all script
13 verbose stack Error: jupyterlab_app@0.1.1 build:all: `npm run build && npm start`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:289:16)
13 verbose stack     at emitTwo (events.js:125:13)
13 verbose stack     at EventEmitter.emit (events.js:213:7)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:40:14)
13 verbose stack     at emitTwo (events.js:125:13)
13 verbose stack     at ChildProcess.emit (events.js:213:7)
13 verbose stack     at maybeClose (internal/child_process.js:927:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
14 verbose pkgid jupyterlab_app@0.1.1
15 verbose cwd /home/marius/src/jupyterlab_app
16 verbose Linux 4.4.0-96-generic
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build:all"
18 verbose node v8.6.0
19 verbose npm  v5.3.0
20 error code ELIFECYCLE
21 error errno 1
22 error jupyterlab_app@0.1.1 build:all: `npm run build && npm start`
22 error Exit status 1
23 error Failed at the jupyterlab_app@0.1.1 build:all script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

--- a/package.json
+++ b/package.json
@@ -48,9 +48,7 @@
         "tar.gz",
         "AppImage"
       ],
-      "desktop": {
-        "Category": "Science;Development;"
-      }
+      "category": "Science;Development;"
     },
     "win": {
       "target": [
@@ -164,13 +162,13 @@
     "@jupyterlab/tooltip": "^0.9.0",
     "@jupyterlab/tooltip-extension": "^0.9.0",
     "@jupyterlab/vega2-extension": "^0.9.0",
-    "@types/react": "^16.0.0",
-    "@types/react-dom": "^15.5.1",
+    "@types/react": "~16.0.19",
+    "@types/react-dom": "~16.0.2",
     "bottlejs": "^1.6.1",
     "electron-debug": "^1.2.0",
     "es6-promise": "^4.1.0",
     "font-awesome": "^4.6.3",
-    "react": "^15.6.1",
-    "react-dom": "^15.6.1"
+    "react": "~16.0.0",
+    "react-dom": "~16.0.0"
   }
 }

--- a/src/browser/app.tsx
+++ b/src/browser/app.tsx
@@ -213,13 +213,13 @@ class Application extends React.Component<Application.Props, Application.State> 
         });
 
         let rServer: Application.IRemoteServer = {...server, id: this._nextRemoteId++};
-        this.setState((prev: ServerManager.State) => {
+        this.setState((prevState: ServerManager.State) => {
             server.id = this._nextRemoteId++;
-            let conns = this.state.remotes.concat(rServer);
+            let remotes = this.state.remotes.concat(rServer);
             this._saveState();
             return({
                 renderState: this._renderEmpty,
-                conns: {servers: conns}
+                remotes: remotes
             });
         });
     }

